lib/Hakyll/Web/Template/Internal.hs

```diff
@@ -1,15 +1,20 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+
 module Hakyll.Web.Template.Internal
-    ( Template (..)
+    ( Template
     , template
+    , unTemplate
+    , getOrigin
```

Comment:
Do we need to export `unTemplate`?  In either case I'd call these `templateElements` and `templateOrigin`.

Reply:
I think we need to export them from `Template.Internal`, yes. About renaming them: `unTemplate` had been used and exported before as part of the non-smart `Template` constructor.